### PR TITLE
[INTEGRATION][AIRFLOW] Use task instance start date for event time of the start event instead of dag execution date.

### PR DESCRIPTION
**Pull request by fm100**
Signed-off-by: Minkyu Park <minkyu@datakin.com>

### Problem
From the airflow 2.x integration, the `eventTime` of the start event uses `dagrun.execution_date` which is scheduled time of the dag but not the actual task start time.

Closes: #455 

### Solution
This PR changes to use `task_instance.start_date` instead of `dagrun.execution_date`.

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)

### CODE REVIEW

File: integration/airflow/openlineage/lineage_backend/__init__.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def send_lineage(
                 run_id=run_id,
                 job_name=job_name,
                 job_description=dag.description,
-                event_time=DagUtils.get_start_time(dagrun.execution_date),
+                event_time=DagUtils.get_start_time(task_instance.start_date),
```
Comment (wslulciuc) [inline integration/airflow/openlineage/lineage_backend/__init__.py:54]: Is there a way we can test the fix?

**Review by wslulciuc — APPROVED** [review]
Minor comment on testing the fix, otherwise 👍

File: integration/airflow/openlineage/lineage_backend/__init__.py (hunk 1 above)
Comment (wslulciuc) [reply integration/airflow/openlineage/lineage_backend/__init__.py:54]: I think one reasonable approach would be to update our Airflow integration test, see [`test_integration.py`](https://github.com/OpenLineage/OpenLineage/blob/main/integration/airflow/tests/integration/test_integration.py). For each DAG run, we call [`wait_for_dag()`](https://github.com/OpenLineage/OpenLineage/blob/main/integration/airflow/tests/integration/test_integration.py#L39) with the DAG ID that queries the `dag_run` table. You can do something similar by checking the start time, nominal times, etc for each DAG run on completion:

```python
check_dag_start_time(DAG_ID)
check_dag_nominal_start_time(DAG_ID)
.
.
```

Comment (fm100) [reply integration/airflow/openlineage/lineage_backend/__init__.py:54]: Thank you for the feedback. I've added a check to the integration test case, simply checking the values and orders of `eventTime` that is actually generated. The issue was not on the DAG information but was the `eventTime` of the openlineage event, so checking the generated event time would be sufficient.

**Pull request merged**